Exiled.Events/Patches/Generic/PlayerVolume.cs

```diff
@@ -0,0 +1,139 @@
+                new(OpCodes.Stloc_S, decoded),
+
+                // plr.VoiceModule.Decoder.Decode(msg.Data, msg.DataLength, decoded);
+                new(OpCodes.Ldloc_S, plr),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(Player), nameof(Player.VoiceModule))),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(VoiceModuleBase), nameof(VoiceModuleBase.Decoder))),
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Ldfld, Field(typeof(VoiceMessage), nameof(VoiceMessage.Data))),
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Ldfld, Field(typeof(VoiceMessage), nameof(VoiceMessage.DataLength))),
+                new(OpCodes.Ldloc_S, decoded),
+                new(OpCodes.Callvirt, Method(typeof(OpusDecoder), nameof(OpusDecoder.Decode), new[] { typeof(byte[]), typeof(int), typeof(float[]) })),
+                new(OpCodes.Pop),
+
+                // for (int i = 0; i < array.Length; i++)
+                new(OpCodes.Ldc_I4_0),
+                new(OpCodes.Stloc_S, pos),
+                // decoded[i] = Mathf.Abs(decoded[i])
+                new(OpCodes.Br_S, loopcheck),
+
+                // loop start
+                new(OpCodes.Ldloc_S, decoded),
+                new(OpCodes.Ldloc_S, pos),
+                new(OpCodes.Ldloc_S, decoded),
+                new(OpCodes.Ldloc_S, pos),
+                new(OpCodes.Ldelem_R4),
+                new(OpCodes.Call, Method(typeof(Mathf), nameof(Mathf.Abs), new[] { typeof(float) })),
+                new(OpCodes.Stelem_R4),
+
+                // i++
+                new(OpCodes.Ldloc_S, pos),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Add),
+                new(OpCodes.Stloc_S, pos),
+
+                // i < array.Length
+                new(OpCodes.Ldloc_S, pos),
+                new(OpCodes.Ldloc_S, decoded),
+                new(OpCodes.Ldlen),
+                new(OpCodes.Conv_I4),
+                new(OpCodes.Blt_S, loopstart),
+
+                // end loop
+
+                // standardVoiceModule.GlobalPlayback.Loudness = array.Sum() / (float)msg.DataLength;
+                new(OpCodes.Ldloc_S, svm),
+                new(OpCodes.Ldfld, Field(typeof(StandardVoiceModule), nameof(StandardVoiceModule.GlobalPlayback))),
+                new(OpCodes.Ldloc_S, decoded),
+                new(OpCodes.Call, Method(typeof(Enumerable), nameof(Enumerable.Sum), new[] { typeof(IEnumerable<float>) })),
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Ldfld, Field(typeof(VoiceMessage), nameof(VoiceMessage.DataLength))),
+                new(OpCodes.Conv_R4),
+                new(OpCodes.Div),
+                new(OpCodes.Callvirt, PropertySetter(typeof(VoiceChatPlaybackBase), nameof(VoiceChatPlaybackBase.Loudness))),
+            });
+
+            newInstructions[index + 26].WithLabels(loopstart);
+            newInstructions[index + 37].WithLabels(loopcheck);
```

Comment:
Use `labels.Add`
And better to again get index

Reply:
Changed Label Management